.github/workflows/foreman_plugin.yml

```
@@ -166,6 +166,11 @@ jobs:
         if: ${{ hashFiles(format('{0}/package.json', inputs.plugin), format('{0}/test/integration/**', inputs.plugin)) != '' && contains(matrix.task, 'test') }}
         run:
           bundle exec rake webpack:compile
+      - name: Archive all_react_app_exports
```

```suggestion
      - name: Archive all_react_app_exports
        if: ${{ hashFiles(format('{0}/package.json', inputs.plugin), format('{0}/test/integration/**', inputs.plugin)) != '' && contains(matrix.task, 'test') }}
```

That way it will only execute for the branches where we actually compiled webpack and the file exists (otherwise you get ugly warnings in the logs)

---

Isnt it an error if webpack compiled and the file doesnt exist?

---

For Foreman? Sure.
But this is the plugin test file, and for plugins that do not have own webpack content, we don't even compile Foremans webpack in CI to save time.

We should add the same archival step in https://github.com/theforeman/foreman/blob/develop/.github/workflows/foreman.yml (and debate whether it should be a fail or not there)

---

https://github.com/theforeman/foreman/pull/10139